src/System.Runtime.Extensions/src/System/BitConverter.cs

```diff
@@ -0,0 +1,512 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+/*============================================================
+**
+** Class:  BitConverter
+**
+**
+** Purpose: Allows developers to view the base data types as 
+**          an arbitrary array of bits.
+**
+** 
+===========================================================*/
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Diagnostics.Contracts;
+using System.Security;
+
+namespace System
+{
+    // The BitConverter class contains methods for
+    // converting an array of bytes to one of the base data 
+    // types, as well as for converting a base data type to an
+    // array of bytes.
+    // 
+    // Only statics, does not need to be marked with the serializable attribute
+    public static class BitConverter
+    {
+        // This field indicates the "endianess" of the architecture.
+        // The value is set to true if the architecture is
+        // little endian; false if it is big endian.
+#if BIGENDIAN
```

Comment:
Some of the files are used only by .NET Native but I've chosen to bring them all over so that everything is at least in one place. Also, we may want to try to use these types for the CoreCLR build in the future (if we can safely pull them from mscorlib).